pkg/apiserver/rest/dualwriter_mode3.go

```diff
@@ -29,106 +30,143 @@ func (d *DualWriterMode3) Mode() DualWriterMode {
 	return Mode3
 }
 
+const mode3Str = "3"
+
 // Create overrides the behavior of the generic DualWriter and writes to LegacyStorage and Storage.
 func (d *DualWriterMode3) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
-	log := klog.FromContext(ctx)
+	log := d.Log.WithValues("kind", options.Kind)
+	ctx = klog.NewContext(ctx, log)
+	var method = "create"
 
+	startStorage := time.Now()
 	created, err := d.Storage.Create(ctx, obj, createValidation, options)
 	if err != nil {
 		log.Error(err, "unable to create object in storage")
+		d.recordLegacyDuration(true, mode3Str, options.Kind, method, startStorage)
 		return created, err
 	}
+	d.recordStorageDuration(false, mode3Str, options.Kind, method, startStorage)
 
-	if _, err := d.Legacy.Create(ctx, obj, createValidation, options); err != nil {
-		log.WithValues("object", created).Error(err, "unable to create object in legacy storage")
-	}
-	return created, nil
+	go func() {
+		ctx, cancel := context.WithTimeoutCause(ctx, time.Second*10, errors.New("legacy create timeout"))
+		defer cancel()
+
+		startLegacy := time.Now()
+		_, errObjectSt := d.Legacy.Create(ctx, obj, createValidation, options)
+		d.recordLegacyDuration(errObjectSt != nil, mode3Str, options.Kind, method, startLegacy)
+	}()
+
+	return created, err
 }
 
 // Get overrides the behavior of the generic DualWriter and retrieves an object from Storage.
 func (d *DualWriterMode3) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
-	return d.Storage.Get(ctx, name, &metav1.GetOptions{})
+	log := d.Log.WithValues("kind", options.Kind)
```

Comment:
Looks like we're missing some form of identification for the object, name for example.